### ISSUE TITLE
fix(solvers): `solve_cg` raises on `max_steps` instead of returning `converged=False`, reports `converged=True` with |r|/|b|≈1 for indefinite operators, and blocks gradients w.r.t. λ/dx by passing the preconditioner through lineax `options`

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/finitevolX#307]
Part of #277. Structural follow-up (adapter over gaussx) in #279.

## Summary

`solve_cg` (`finitevolx/_src/solvers/iterative.py:48-136`) wraps `lineax.CG` and returns a `CGInfo(iterations, residual_norm, converged)`. Three defects:

1. `lx.linear_solve` is called without `throw=False` (`:127`), so hitting `max_steps` or a NaN **raises** `EquinoxRuntimeError`; `converged=False` is never returned.
2. `converged` is `True` after 0–1 iterations with |r|/|b| ≈ 0.8–1 when the operator is indefinite (λ < 0) or the preconditioner has the wrong sign — lineax's early exit on a non-decreasing residual is reported as success.
3. The preconditioner is passed via `options={"preconditioner": ...}`, and lineax refuses to differentiate through `options`, so `jax.grad` w.r.t. `lambda_` or `dx` through `method="cg"` fails (`lineax.linear_solve(..., options=...) cannot be autodifferentiated`); without a preconditioner the gradients work and match finite differences.

## Reproduction

```
solve_cg(max_steps=3):                           RAISES EquinoxRuntimeError (expected converged=False)
indefinite λ = -50:                              iters=1, residual=8.9 (|b|=11.5), converged=True
wrong-sign preconditioner:                       iters=0, residual=11.5, converged=True
grad w.r.t. lambda_ through method="cg"+fft pc:  FAIL (options) ; without pc: OK, matches FD
```

(Scripts: audit `fvx_solvers/cg_stall.py`, `jax_compat.py`.)

## Proposed fix

- `throw=False`; `converged = (result == lx.RESULTS.successful) & (residual ≤ atol + rtol·|b|)` computed from an explicit final residual.
- Pass the preconditioner as a `lineax.AbstractLinearOperator` (`FunctionLinearOperator` with the sign flip folded in) via the solver's own preconditioner mechanism, or go through `gaussx.linear_solve(..., preconditioner=OperatorPreconditioner(...))`, which is differentiable and reproduces `solve_cg` bit-for-bit (see #279).
- Document that raw `solve_cg` output is non-zero outside the mask (preconditioner leakage) and that the wrappers mask it.

## Acceptance criteria

- [ ] `max_steps=3` returns `converged=False` without raising.
- [ ] λ < 0 returns `converged=False`.
- [ ] `jax.grad` w.r.t. `lambda_` and `dx` through `method="cg"` with the FFT preconditioner matches central finite differences to 1e-6.
